crates/wasmtime/src/runtime/vm/component/states.rs

```diff
@@ -0,0 +1,134 @@
+use {
```

Comment:
This file was shamelessly copied from resources.rs and modified for use with stream/future/error-context state.  Those handles are different enough from resource handles that I didn't want to try to address everything with a single data structure.  I'm open to other approaches, though.

Reply:
Could you expand the documentation for this struct as to what problem it's solving? For example what extra functionality it has over `ResourceTable`. I'm a bit surprised for example of the `reps_to_indexes` reverse mapping where most of the component model stuff doesn't require hash maps today and that seems like it's bordering on needing something like that.

Also this is something I probably should have done for `ResourceTable` but if it's possible this'd be a good place to have some `#[test]` unit tests for this data structure doing some simple pushes/pops/etc.